clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -242,9 +254,16 @@ func (w *waitForControlledPodsRunningMeasurement) gather(syncTimeout time.Durati
 			numberRunning++
 		case deleted:
 			numberDeleted++
-		case timeout:
+		case deleteTimeout, timeout:
 			timedOutObjects = append(timedOutObjects, objChecker.key)
 			numberTimeout++
+
+			if status == deleteTimeout {
+				err := k8sClient.CoreV1().Pods(w.selector.Namespace).Delete(context.TODO(), objChecker.key, forceDeletePod)
```

Comment:
objChecker.key isn't the pod name, I think.

You probably need to use DeleteCollection here and pass appropriate selectors...

Reply:
ok.. let me check how to get the pod name. And also about the selector usage with DeleteCollection

Reply:
You want DeleteCollection - looking for pod name isn't the right way to go.